optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/common/GenuineRuleMetadata.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.drools.common;
+
+import org.drools.model.Declaration;
+import org.drools.model.PatternDSL;
+
+public final class GenuineRuleMetadata<A> implements RuleMetadata<A> {
+
+    private final Declaration<A> variableDeclaration;
+    private final PatternDSL.PatternDef<A> pattern;
+
+    GenuineRuleMetadata(Declaration<A> variableDeclaration, PatternDSL.PatternDef<A> pattern) {
```

Comment:
code style: we don't use friendly constructors. either private, protected or public

Reply:
This is one of the least understandable rules around the codebase.

In this case, it's going to have one of the following results:

1. The class gets a public constructor, even though nobody outside of this package has any business creating instances.
1. The class gets hidden inside another class, emulating friendly access and making the code worse readable at the same time.

Reply:
I consider using the most restrictive access modifiers as possible a rule of thumb. In my opinion, the code style convention brings little value for the price of preventing us from using the package private access.